File: ruby/features/step_definitions/data-in-steps-step-example.rb
Given("I have {string} addresses") do |address|
    puts address
end

When("I add my {string} address") do |addresses|
    puts addresses
end

Then("I will have {string} addresses") do |address|
    puts address
end

Given(/^I have a ([a-z]+)$/) do |fruit|
    puts fruit
end

When("I take an apple") do
    # When glue code here
end

Then(/^I have an ([a-z]+) and a ([a-z]+)$/) do |fruit1, fruit2|
    puts fruit1
    puts fruit2
end

Given("I have divided the number {int} in half") do |integer|
    puts integer
end

Then("I will have a result of {float}") do |float|
    puts float
end

File: javascript/features/support/data-in-steps-step-example.js
const { Given, When, Then } = require('cucumber')

Given('I have {string} addresses', function (address) {
    console.log(address);
});

When('I add my {string} address', function (addresses) {
    console.log(addresses);
});

Then('I will have {string} addresses', function (address) {
    console.log(address);
});

Given(/^I have a ([a-z]+)$/, function (fruit) {
    console.log(fruit);
});

When('I take an apple', function () {
    // When glue code here
});

Then(/^I have an ([a-z]+) and a ([a-z]+)$/, function (fruit1, fruit2) {
    console.log(fruit1);
    console.log(fruit2);
});

Given('I have divided the number {int} in half', function (integer) {
    console.log(integer);
});

Then('I will have a result of {float}', function (float) {
    console.log(float);
});